cmd/demoserver/kvdatamake.go

```diff
@@ -195,6 +198,32 @@ func (*DataRWServer) ListFiles(ctx context.Context, folder *pb.ListFilesReq) (*p
 	return &pb.ListFilesResponse{FileNames: files}, nil
 }
 
+func (s *DataRWServer) IsReady(ctx context.Context, req *pb.IsReadyRequest) (*pb.IsReadyResponse, error) {
+	select {
+	case <-s.ready:
+		return &pb.IsReadyResponse{Ready: true}, nil
+	default:
+		return &pb.IsReadyResponse{Ready: false}, nil
+	}
+}
+
+func (s *DataRWServer) CheckDir(ctx context.Context, req *pb.CheckDirRequest) (*pb.CheckDirResponse, error) {
+	dir := req.Dir
+	for path := range s.fileWriterMap {
+		name := filepath.Base(path)
+		destPath := filepath.Join(dir, name)
+		cmd := exec.Command("diff", path, destPath)
+		stdout, err := cmd.Output()
```

Comment:
Better to use a golang library, such as `https://github.com/sergi/go-diff`, in case of `diff` is not available (no binary or access path is restricted)